builtin/providers/azurerm/resource_arm_template_deployment.go

```diff
@@ -155,20 +155,44 @@ func resourceArmTemplateDeploymentRead(d *schema.ResourceData, meta interface{})
 	if resp.Properties.Outputs != nil && len(*resp.Properties.Outputs) > 0 {
 		outputs = make(map[string]string)
 		for key, output := range *resp.Properties.Outputs {
+			log.Printf("[DEBUG] Processing deployment output %s", key)
 			outputMap := output.(map[string]interface{})
 			outputValue, ok := outputMap["value"]
 			if !ok {
 				// No value
 				continue
 			}
+			outputType := outputMap["type"]
+			var outputStringValue string
+
+			switch outputType {
+			case "Bool":
+				if outputValue == false {
+					outputStringValue = "0"
+				} else if outputValue == true {
+					outputStringValue = "1"
+				} else {
+					return fmt.Errorf("Invalid value %s for boolean output %s", outputValue, key)
+				}
+
+			case "String": // Nothing to do
+				fallthrough
+			case "Int":
+				outputStringValue = fmt.Sprint(outputValue)
+
+			case "SecureString", "Object", "SecureObject", "Array":
+				fallthrough
+			default:
+				log.Printf("[WARNING] Ignoring output %s: Outputs of type %s are not currently supported in azurerm_deployment_template.",
```

Comment:
Also, is there a way to actual print an actual warning in the main Terraform output (as opposed to just the debugging output)?  I originally returned an error here, but that caused problems because the outputs still ended up in the state file and continued causing errors even after the user corrected the problem).  Haven't investigated how to fix that and return an actual error.